File: October2022/Python/BreakAPalindrome.py
# https://leetcode.com/problems/break-a-palindrome/
# Author : @Jiganesh

class Solution:
    
    # TC : O(N)
    # SC : O(N)
    def breakPalindrome(self, palindrome: str) -> str:

        palindrome = list(palindrome)
        
        n = len(palindrome)
        mid = n //2

        if n < 2: return ""

        for i in range (mid):
            if palindrome[i]!= "a":
                palindrome[i]="a"
                return "".join(palindrome)

        palindrome[-1]="b"
        return "".join(palindrome)                

File: October2022/Java/BreakAPalindrome.java
// @saorav21994

class Solution {
    public String breakPalindrome(String palindrome) {
        int l = palindrome.length();
        if (l == 1)
            return "";
        else {
            char [] ch = palindrome.toCharArray();
            for (int i = 0; i < l/2; i++) {
                if (ch[i] == ch[l-i-1]) {
                    if (ch[i] != 'a') {
                        ch[i] = 'a';
                        return new String(ch);
                    }
                    
                }
            }
            ch[l-1] = 'b';
            return new String(ch);
        }
    }
}
